src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java

```diff
@@ -125,23 +125,34 @@ EurekaOneDiscoveryStrategy build() {
     private static final int DISCOVERY_RETRY_TIMEOUT = 1;
 
     private final EurekaClient eurekaClient;
+    private final String groupName;
     private final ApplicationInfoManager applicationInfoManager;
 
     private final Boolean useClasspathEurekaClientProps;
     private final String namespace;
     private StatusChangeStrategy statusChangeStrategy;
+    private final Boolean skipEurekaRegistrationVerification;
+    private final Boolean useMetadataForHostAndPort;
 
     private EurekaOneDiscoveryStrategy(final EurekaOneDiscoveryStrategyBuilder builder) {
         super(builder.logger, builder.properties);
 
         this.namespace = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, NAMESPACE, "hazelcast");
         boolean selfRegistration = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, SELF_REGISTRATION, true);
+        this.useMetadataForHostAndPort = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, USE_METADATA_FOR_HOST_AND_PORT, false);
+        this.skipEurekaRegistrationVerification = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, SKIP_EUREKA_REGISTRATION_VERIFICATION, false);
         this.useClasspathEurekaClientProps = getOrDefault(EUREKA_ONE_SYSTEM_PREFIX, USE_CLASSPATH_EUREKA_CLIENT_PROPS, true);
-        // override registration if requested
-        if (!selfRegistration) {
+        this.groupName = builder.groupName != null ? builder.groupName : GroupConfig.DEFAULT_GROUP_NAME;
+        if (builder.changeStrategy != null) {
+            this.statusChangeStrategy = builder.changeStrategy;
```

Comment:
Wasn't the original logic to only use `builder.changeStrategy` as a last resort?
In other words, the `selfRegistration` flag was supposed to override a custom `changeStrategy`. Following the same logic, the `useMetadataForHostAndPort` should do the same. But with this check here we ignore the flags if a change strategy is defined with the builder... What do you think?

Reply:
_Wasn't the original logic to only use `builder.changeStrategy` as a last resort?_

Probably that was the idea. But I can't explain the reason behind that. Personally I prefer current solution. When someone (usually in unit tests) provides `StatusChangeStrategy` it will be used. When strategy is not provided, we choose strategy using flags.

What do you think? I don't want to break anything with this change.

Reply:
I'm not against it, but it would be a change in semantics, i.e., potentially breaking. I suggest we keep the old behavior here, and perhaps keep track of this discussion in a separate issue.

Reply:
OK. This change was reverted. I will create separate ticket after we finish this PR.